lib/normalizer.c

```diff
@@ -3428,6 +3428,57 @@ grn_nfkc_normalize_unify_romaji(grn_ctx *ctx,
                                     n_unified_characters);
 }
 
+static const unsigned char *
+grn_nfkc_normalize_unify_iteration_mark(grn_ctx *ctx,
+                                        const unsigned char *start,
+                                        const unsigned char *current,
+                                        const unsigned char *end,
+                                        size_t *n_used_bytes,
+                                        size_t *n_used_characters,
+                                        unsigned char *unified_buffer,
+                                        size_t *n_unified_bytes,
+                                        size_t *n_unified_characters,
+                                        void *user_data)
+{
+  size_t char_length;
+
+  char_length = (size_t)grn_charlen_(ctx, current, end, GRN_ENC_UTF8);
+
+  *n_used_bytes = char_length;
+  *n_used_characters = 1;
+
+  /* U+3005 IDEOGRAPHIC ITERATION MARK 々 */
+  if (char_length == 3 && current[0] == 0xe3 && current[1] == 0x80 &&
+      current[2] == 0x85) {
+    if (current > start) {
+      const unsigned char *prev = current - 1;
+
+      while (prev > start && (*prev & 0xc0) == 0x80) {
+        prev--;
+      }
```

Comment:
In general, we should not backward characters in UTF-8. We can detect the first byte by the top 2 bits but it has overhead like this.

Can we keep the previous character length in `user_data` like `grn_nfkc_normalize_unify_kana_prolonged_sound_mark_like()` does?